pulp_rpm/app/migrations/0052_modulemd_digest.py

```diff
@@ -0,0 +1,51 @@
+# Generated by Django 4.2.4 on 2023-09-04 14:06
+
+from django.db import migrations, models
+import hashlib
+
+
+def add_snippet_hash(apps, schema_editor):
+    """Calculate and add digest hash of the snippet."""
+
+    Modulemd = apps.get_model("rpm", "Modulemd")
+    modules_to_update = []
+
+    for mmd in Modulemd.objects.all().only("snippet").iterator():
+        digest = hashlib.sha256(mmd.snippet.encode()).hexdigest()
```

Comment:
@ggainey @dralley previous 2 migrations where written to account for the case when a module would have an empty snippet. Is this still the case we need to account for? In that case I'd need to re-write this part a bit, we will however still have a very slim chance for a migration to fail in case there would be two modules with the same NSVCA and both wold be having empty snippets because I will be using digest of an empty blob `e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855`

Reply:
It depends.  It is hopefully less needed than it was at the time, but it might still be needed.  This was part of why we were resaving module snippets in the first place.  Ideally if that has been in place for a little while it will have been self-repairing.

On the other hand if there's one module with an empty snippet there could just as easily be more than one.

Maybe it's better to fail out early in this case w/ a report about what is wrong, and defer to data repair?

Reply:
I was saying nonsense, given our current constraints we cannot have in DB 2 same NSVCA + empty snippet. When this migration will be running it will always be only one ( unique NSVCA + empty snippet).

I'd probably do nothing at this point  since empty snippet can be fixed by re-running the sync, so then the old module will be removed and new module with -non-empty-snippet will be saved and added to the new RV

Reply:
>I'd probably do nothing at this point since empty snippet can be fixed by re-running the sync, so then the old module will be removed and new module with -non-empty-snippet will be saved and added to the new RV

In theory, as long as the module hasn't disappeared, which IIRC Fedora was doing for a while.  That is, older versions of streams would be purged.  Or maybe I'm thinking of EPEL, but I remember the modularity group wasn't thrilled about it.

I'm not sure if this is worth worrying about *yet* though.  Maybe it'll be fine?